zallet/src/config.rs

```diff
@@ -35,34 +46,117 @@ pub struct ZalletConfig {
+    #[serde(default)]
     pub external: ExternalSection,
 
     /// Settings for Zallet features.
+    #[serde(default)]
     pub features: FeaturesSection,
 
     /// Settings for the Zaino chain indexer.
+    #[serde(default)]
     pub indexer: IndexerSection,
 
     /// Settings for the key store.
+    #[serde(default)]
     pub keystore: KeyStoreSection,
 
     /// Settings for how Zallet manages notes.
+    #[serde(default)]
     pub note_management: NoteManagementSection,
 
     /// Settings for the JSON-RPC interface.
+    #[serde(default)]
     pub rpc: RpcSection,
 }
 
+/// Zallet Configuration Filename
+pub const CONFIG_FILE: &str = "zallet.toml";
+
 impl ZalletConfig {
+    /// Loads Zallet configuration from conventional sources.
+    ///
+    /// Configuration is loaded from three sources, in order of precedence:
+    /// 1. Hard-coded defaults (lowest precedence)
+    /// 2. TOML configuration file (if provided)
+    /// 3. Environment variables with `ZALLET_` prefix (highest precedence)
+    ///
+    /// Environment variables use the format `ZALLET_SECTION__KEY` where:
+    /// - `SECTION` is the configuration section (e.g., `builder`, `rpc`)
+    /// - `KEY` is the configuration key within that section
+    /// - Double underscores (`__`) separate nested keys
+    ///
+    /// # Examples
+    /// - `ZALLET_BUILDER__SPEND_ZEROCONF_CHANGE=true` sets `builder.spend_zeroconf_change = true`
+    /// - `ZALLET_RPC__BIND=127.0.0.1:28232,127.0.0.1:28233` sets multiple bind addresses (comma-separated)
+    /// - `ZALLET_RPC__BIND=["127.0.0.1:28232"]` also works (JSON array format)
+    /// - `ZALLET_RPC__TIMEOUT=30` sets `rpc.timeout = 30`
+    ///
+    /// # Security
+    /// Environment variables whose leaf key names end with sensitive suffixes (case-insensitive)
+    /// are blocked from overriding config values: `password`, `secret`, `token`, `cookie`, `private_key`.
+    pub fn load(config_path: Option<&Path>) -> Result<Self, config::ConfigError> {
+        let mut builder = config::Config::builder();
+
+        // Add config file if provided
+        if let Some(path) = config_path {
+            builder = builder.add_source(config::File::from(path).required(true));
+        }
+
+        // Load from standard ZALLET_ environment variables with a sensitive-leaf deny-list
+        let mut filtered_env: HashMap<String, String> = HashMap::new();
+        for (key, value) in env::vars() {
+            if let Some(stripped) = key.strip_prefix("ZALLET_") {
+                // Extract the leaf key (rightmost part after splitting on __)
+                let leaf_key = stripped.split("__").last().unwrap_or(stripped);
+                if !is_sensitive_leaf_key(leaf_key) {
+                    filtered_env.insert(key, value);
+                }
+                // Sensitive keys are silently ignored
+            }
+        }
+
+        // Add filtered environment variables (highest precedence)
+        builder = builder.add_source(
```

Comment:
In the group chat, we discussed only supporting env-var setting in Docker images specifically. If we take that approach, this would be where we configure it:
```suggestion
        // Add filtered environment variables (highest precedence)
        #[cfg(zallet_docker_image)]
        builder = builder.add_source(
```